app/Models/Rating.php

```diff
@@ -48,4 +62,92 @@ public function author()
     {
         return $this->belongsTo(\App\Models\User::class);
     }
+
+    /**
+     * Determines if the rating instance has a category given by the passed parameter
+     * @param $bitMask
+     * bitmask param should always be one of the bitmask constants defined on this model
+     * @return bool
+     */
+    public function hasCategory($bitMask)
+    {
+        return ($this->categories & $bitMask) === $bitMask;
+    }
+
+    /**
+     * Adds a category given by the bitmask parameter, which should be one of the bitmask constants defined on this model
+     * @param $bitMask
+     */
+    public function addCategory($bitMask)
+    {
+        $this->categories |= $bitMask;
+    }
+
+    /**
+     * Removes a category given by the bitmask parameter, which should be one of the bitmask constants defined on this model
+     * @param $bitMask
+     */
+    public function removeCategory($bitMask)
+    {
+        $this->categories &= ~$bitMask;
+    }
+
+    /**
+     * Returns category labels. Not recommended for large batches of requests, instead use hasCategory with bitmask
+     * @returns Collection of strings corresponding to the rating categories of this entity
+     */
+    public function getCategoryLabels()
+    {
+        $labels = collect();
+        $bitmasks = collect([
+            self::CREATIVE_BITMASK,
+            self::FLEXIBLE_BITMASK,
+            self::FRIENDLY_BITMASK,
+            self::HELPFUL_BITMASK,
+            self::PREPARED_BITMASK,
+        ]);
+
+        foreach ($bitmasks as $mask) {
+            if ($this->hasCategory($mask)) {
+                $labels->push($this->getCategoryLabelFromMask($mask));
+            }
+        }
+
+        return $labels;
+    }
+
+    /**
+     * Get category label corresponding to a bitmask
+     * @param $mask integer The bitmask
+     * @return string The Label
+     */
+    private function getCategoryLabelFromMask($mask)
```

Comment:
theoretically this whole method can go away if you add the changes on Line 99

Reply:
removed it